src/components/TopicList.jsx

```diff
@@ -0,0 +1,41 @@
+import React, { useState, useEffect } from "react";
+import { Link } from "react-router-dom";
+import axios from "axios";
+
+const TopicList = () => {
+  const [topics, setTopics] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    axios
+      .get("https://cwazycodes-nc-news.onrender.com/api/topics")
+      .then((response) => {
+        setTopics(response.data.topics);
+        setLoading(false);
+      })
+      .catch((err) => {
+        console.erro(err);
+        setError("Failed to load topics. Please try again.");
+        setLoading(false);
+      });
+  }, []);
+
+  if (loading) return <p>Loading topics...</p>;
+  if (error) return <p>{error}</p>;
+
+  return (
+    <div className="topic-list">
+      <h2>Topics</h2>
+      <ul>
+        {topics.map((topic) => (
+          <li key={topic.slug}>
+            <Link to={`/topics/${topic.slug}`}>{topic.slug}</Link>
+          </li>
```

Comment:
You have set your topics to appear in a list. That is ok when you are dealing with only 3 topics. But how would this work when you have  100+ topics?

There are better ways of displaying topics eg. a drop down, perhaps on the article page instead of its own?

Food for thought!